tools/c7n_azure/c7n_azure/resources/entraid_group.py

```diff
@@ -0,0 +1,518 @@
+
+    def get_graph_resources(self):
+        """Get resources from Microsoft Graph API for use with GraphSource."""
+        try:
+            response = self.make_graph_request('groups')
+            resources = response.get('value', [])
+
+            log.debug(f"Retrieved {len(resources)} groups from Graph API")
+
+            # Augment resources with additional computed fields
+            resources = self.augment(resources)
+
+            log.debug(f"Returning {len(resources)} groups after augmentation")
+            return resources
+        except requests.exceptions.RequestException as e:
+            log.error(f"Error retrieving EntraID groups: {e}")
+            if "Insufficient privileges" in str(e) or "403" in str(e):
+                log.error(
+                    "Insufficient privileges to read groups. "
+                    "Required permissions: Group.Read.All"
+                )
+            return []
+
+    def augment(self, resources):
+        """Augment group resources with additional Graph API data"""
+        try:
+            # Enhance with additional properties
+            for resource in resources:
+                # Add computed fields for policy evaluation
+                resource['c7n:IsSecurityGroup'] = self._is_security_group(resource)
+                resource['c7n:IsDistributionGroup'] = self._is_distribution_group(resource)
+                resource['c7n:IsDynamicGroup'] = self._is_dynamic_group(resource)
+                resource['c7n:IsAdminGroup'] = self._is_admin_group(resource)
+
+        except Exception as e:
+            log.warning(f"Failed to augment EntraID groups: {e}")
+
+        return resources
+
+    def _is_security_group(self, group):
+        """Determine if group is a security group"""
+        return (
+            group.get('securityEnabled', False) and
+            not group.get('mailEnabled', False)
+        )
+
+    def _is_distribution_group(self, group):
+        """Determine if group is a distribution group"""
+        return group.get('mailEnabled', False)
+
+    def _is_dynamic_group(self, group):
+        """Determine if group uses dynamic membership"""
+        group_types = group.get('groupTypes', [])
+        return 'DynamicMembership' in group_types
+
+    def _is_admin_group(self, group):
+        """Determine if group has administrative privileges"""
+        display_name = group.get('displayName', '').lower()
+        admin_indicators = [
+            'admin', 'administrator', 'global', 'privileged',
+            'security', 'compliance', 'exchange', 'sharepoint'
+        ]
+        return any(indicator in display_name for indicator in admin_indicators)
+
+    def get_group_member_count(self, group_id):
+        """Get accurate member count for a group using Graph API.
+
+        Required permission: GroupMember.Read.All
+        """
+        try:
+            # Use $count parameter for efficient counting
+            endpoint = f'groups/{group_id}/members/$count'
```

Comment:
Looks like when we're using `$count`, we need to add a `ConsistencyLevel: eventual` header ([reference](https://learn.microsoft.com/en-us/graph/aad-advanced-queries)) to avoid errors like the one I hit in local testing:

```console
Raw count ($count) is not supported because
the 'ConsistencyLevel:eventual' header is missing. Refer to
https://aka.ms/graph-docs/advanced-queries for more
information
```